Mathlib/Topology/CWComplex/Classical/Finite.lean

```diff
@@ -0,0 +1,374 @@
+/-
+Copyright (c) 2025 Floris van Doorn and Hannah Scholz. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Floris van Doorn, Hannah Scholz
+-/
+
+import Mathlib.Topology.CWComplex.Classical.Basic
+
+/-!
+# Finiteness notions on CW complexes
+
+In this file we define what it means for a CW complex to be finite dimensional, of finite type or
+finite. We define constructors with relaxed conditions for CW complexes of finite type and
+finite CW complexes.
+
+## Main definitions
+* `FiniteDimensional`: A CW complex is finite dimensional if it has only finitely many
+  nonempty indexing types for the cells.
+* `FiniteType`: A CW complex is of finite type if it has only finitely many cells in each dimension.
+* `Finite`: A CW complex is finite if it is finite dimensional and of finite type.
+
+## Main statements
+* `RelCWComplex.mkFiniteType`: If we want to construct a CW complex of finite type, we can relax the
+  condition `mapsTo`.
+* `RelCWComplex.mkFinite`: If we want to construct a finite CW complex, we can relax the condition
+  `mapsTo` and can leave out he condition `closed'`.
+* `RelCWComplex.finite_iff_finite_cells`: A CW complex is finite iff the total number of cells is
+  finite.
+-/
+
+open Metric Set
+
+namespace Topology
+
+/-- A CW complex is finite dimensional if `cell C n` is empty for all but finitely many `n`.-/
+class RelCWComplex.FiniteDimensional.{u} {X : Type u} [TopologicalSpace X] (C : Set X) {D : Set X}
+    [RelCWComplex C D] : Prop where
+  /-- For some natural number `n` the type `cell C m` is empty for all `m ≥ n`.-/
```

Comment:
I'd write this as:
```suggestion
/-- For some natural number `n`, the type `cell C m` is empty for all `m ≥ n`. -/
```
Note the space at the end: when you merge master, a linter will enforce this - so please check the other doc-strings as well.